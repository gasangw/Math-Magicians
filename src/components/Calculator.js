import React from "react";
import "./Layout.css"
import calculate from '../logic/calculate';
class Structure extends React.Component {
  constructor(props){
    super(props)
    this.state = {
      total: 0,
      next: '',
      operation: '',
    }
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick = (event) => {
    const buttonName = event.target.value;
    this.setState((prevState) => calculate(prevState, buttonName));
  }

  render(){
  return (
    <div className="frame">
      <div className="calculator">
        <br />
        <div> 
        <button className="screen">{this.state.total}{this.state.operation}{this.state.next}</button>      
        </div>
        <div className="keys">
          <input
            type="button"
            value="AC"
            className="button-clear"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="+/-"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="%"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="&#247;"
            className="button-operator"
            id="red"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="7"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="8"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="9"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="x"
            className="button-operator"
            id="red"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="4"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="5"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="6"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="-"
            className="button-operator"
            id="red"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="1"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="2"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="3"
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="+"
            className="button-operator"
            id="red"
            onClick={this.handleClick}
          />

          <input
            type="button"
            value="0"
            className="button-operator"
            id="blue"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="."
            className="button-operator"
            onClick={this.handleClick}
          />
          <input
            type="button"
            value="="
            className="button-operator"
            id="red"
            onClick={this.handleClick}
          />
        </div>
      </div>
    </div>
  );
  }
};

export default Structure;
