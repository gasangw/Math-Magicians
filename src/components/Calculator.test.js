import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('Quote renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});