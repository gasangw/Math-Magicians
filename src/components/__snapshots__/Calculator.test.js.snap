// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders correctly 1`] = `
<div
  className="frame"
>
  <h3
    className="math"
  >
    Let's do some Math!
  </h3>
  <div
    className="calculator"
  >
    <br />
    <div>
      <button
        className="screen"
      />
    </div>
    <div
      className="keys"
    >
      <input
        className="button-clear"
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="button-operator"
        id="red"
        onClick={[Function]}
        type="button"
        value="รท"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="button-operator"
        id="red"
        onClick={[Function]}
        type="button"
        value="x"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="button-operator"
        id="red"
        onClick={[Function]}
        type="button"
        value="-"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        className="button-operator"
        id="red"
        onClick={[Function]}
        type="button"
        value="+"
      />
      <input
        className="button-operator"
        id="blue"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        className="button-operator"
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="button-operator"
        id="red"
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </div>
</div>
`;
